### PR TITLE
splash page join button works with new join flow

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Step towards resolving https://github.com/LLK/scratch-www/issues/3053

### Changes:

makes join button on splash page either open existing join flow, or (if flag set to true) navigate user to /join.

### CODE REVIEW

File: src/components/intro/intro.jsx [hunk 1]
```diff
@@ -15,12 +15,20 @@ class Intro extends React.Component {
...
         super(props);
         bindAll(this, [
+            'handleClickRegistration',
             'handleShowVideo'
         ]);
         this.state = {
             videoOpen: false
         };
     }
+    handleClickRegistration (event) {
```
Comment (picklesrus) [inline src/components/intro/intro.jsx]: Is there a way to share this and navigateToRegistration between navigation.js and here? It'd be nice if the code were shared so we don't run into this again.

Comment (benjiwheeler) [reply src/components/intro/intro.jsx]: Hm, I think you're right, that would be better.

I rewrote the whole thing to refactor that logic into the reducer. take a look.

File: test/unit/redux/navigation.test.js [hunk 2]
```diff
@@ -238,4 +247,28 @@ describe('unit test lib/validate.js', () => {
...
+            registrationOpen: false,
+            useScratch3Registration: true
+        };
+        const action = handleRegistrationRequested();
+        const resultState = navigationReducer(initialState, action);
+        expect(resultState.registrationOpen).toBe(false);
+        expect(global.window.location).toEqual('/join');
+    });
+
+    test('handleRegistrationRequested with useScratch3Registration false does NOT navigate user away, ' +
```
Comment (picklesrus) [inline test/unit/redux/navigation.test.js]: Yay tests!  What about the case where registrationOpen is set to true and so is useScratch3Registration?  I think it is worth making sure we have a test for that case too.

Comment (benjiwheeler) [reply test/unit/redux/navigation.test.js]: Good idea, done!

**Comment by benjiwheeler** [comment]
Tests are performing differently in Travis vs. local... not sure why!

**Pull request merged**